core/services/address/CountrySubRegionDao.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace EventEspresso\core\services\address;
+
+use EE_Country;
+use EE_Error;
+use EEM_State;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+use EventEspresso\core\exceptions\InvalidInterfaceException;
+use InvalidArgumentException;
+use ReflectionException;
+use stdClass;
+
+/**
+ * Class CountrySubRegionDao
+ * Data Access Object for retrieving Country and Country SubRegion data
+ *
+ * @package EventEspresso\core\services\address
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class CountrySubRegionDao
+{
+
+    const REPO_URL = 'https://raw.githubusercontent.com/eventespresso/countries-and-provinces-states-regions/master/';
+
+    /**
+     * @var array $countries
+     */
+    private $countries = array();
+
+
+    /**
+     * @param EE_Country $country_object
+     * @return void
+     * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws ReflectionException
+     */
+    public function saveCountrySubRegions(EE_Country $country_object)
+    {
+        $CNT_ISO = $country_object->ID();
+        $has_sub_regions = EEM_State::instance()->count(array(array('Country.CNT_ISO' => $CNT_ISO)));
```

Comment:
should models be brought in as a dependency in the constructor rather than used directly?

Reply:
now THAT should absolutely be the case... ug can't believe I missed that.. thnx